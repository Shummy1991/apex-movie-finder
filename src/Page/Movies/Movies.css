.movie-image {
    width: 100%;
    border: 1px solid white;
}

.movie-card-content {
    background-color: #212e3c;
    padding-bottom: 16px !important;
}

h4 {
    margin-top: 0;
    margin-bottom: 12px;
    font-size: 21px;
    cursor: pointer;
}

.movie-genres {
    font-style: italic;
    margin-top: 12px;
    font-size: 16px;
    margin-bottom: 12px;
}

.movie-score {
    font-size: 15px;
}
