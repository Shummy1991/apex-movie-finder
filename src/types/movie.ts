export interface IGenre {
    name: string;
}

export interface IMovie {
    id: string;
    name: string;
    genres: IGenre[];
    score: number;
    overview: string;
    socialMedia: {
        imdb: string | null;
    };
    img: {
        url: string;
    } | null;
}
