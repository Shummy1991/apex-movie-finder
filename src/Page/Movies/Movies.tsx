import { Card, CardContent, Grid, Stack } from "@mui/material";
import { useState } from "react";

import MovieDetails from "../MovieDetails";
import { IMoviesProps } from "./Movies.models";
import { getGenres } from "./Movies.utils";
import "./Movies.css";

const Movies = ({ movies, handleGetRelatedMovies }: IMoviesProps) => {
    // The original idea was to display wiki below the card, but there were too much space left on the right side.
    const [selectedMovie, setSelectedMovie] = useState<string | null>(null);

    return (
        <Stack spacing={2}>
            {movies.map(movie => (
                <Card key={movie.id}>
                    <CardContent className="movie-card-content">
                        <Grid container spacing={2}>
                            <Grid item xs={4}>
                                <img src={movie.img?.url} className="movie-image" />
                            </Grid>
                            <Grid item xs={8}>
                                <h4 onClick={() => setSelectedMovie(movie.id)}>
                                    {movie.name}
                                </h4>
                                <div className="movie-genres">
                                    {getGenres(movie.genres)}
                                </div>
                                <div className="movie-score">
                                    Score: {movie.score}
                                </div>
                                {selectedMovie === movie.id && (
                                    <MovieDetails movie={movie} handleGetRelatedMovies={handleGetRelatedMovies} />
                                )}
                            </Grid>
                        </Grid>
                    </CardContent>
                </Card>
            ))}
        </Stack>
    )
};

export default Movies;
