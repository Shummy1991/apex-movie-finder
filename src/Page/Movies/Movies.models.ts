import { IMovie } from "../../types";

export interface IMoviesProps {
    movies: IMovie[];
    handleGetRelatedMovies: (id: string) => void;
}
