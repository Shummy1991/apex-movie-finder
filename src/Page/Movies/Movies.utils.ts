import { IGenre } from "../../types";

export const getGenres = (genres: IGenre[]) => {
    return genres
        .map(genre => genre.name)
        .join(", ");
};
