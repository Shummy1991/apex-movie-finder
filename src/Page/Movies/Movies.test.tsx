import "jest";
import { fireEvent, render } from "@testing-library/react";

import { mockMovie } from "../../test/movie";

import Movies from "./Movies";

const defaultProps = {
    movies: [mockMovie],
    handleGetRelatedMovies: jest.fn(),
};

afterAll(() => {
    jest.clearAllMocks();
})

describe("MovieDetails", () => {
    it("matches snapshot", () => {
        const { asFragment } = render(<Movies {...defaultProps} />)
        const component = asFragment();
        expect(component).toMatchSnapshot();
    });

    it("title is clickable", async () => {
        const { findByText } = render(<Movies {...defaultProps} />)
        const component = await findByText(mockMovie.name);
        fireEvent.click(component);
    });
});
