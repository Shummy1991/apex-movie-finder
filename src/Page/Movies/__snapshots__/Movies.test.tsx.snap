// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1p5q5e5-MuiStack-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root movie-card-content css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <img
              class="movie-image"
              src="https://myimage.com/image.jpg"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
          >
            <h4>
              The batman
            </h4>
            <div
              class="movie-genres"
            >
              genre 1, genre 2
            </div>
            <div
              class="movie-score"
            >
              Score: 5.4
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
